### ISSUE TITLE
REGRESSION - Graph repository fails in containers (observed in CTS) on 2.2 release - k8s

### ISSUE THREAD

**Issue by planetf1** [odpi/egeria#3560]
When testing the 2.2 release I noted:
 * CTS using the inmemory repository works perfectly
 * CTS fails using the graph repository
 * failure occurs on 2 different k8s environments
 * Log files are posted to https://ibm.box.com/s/isiprgojsi3dbow6n7lswhvqf9spjojn including the 'dev.log' which is the stdout/err from the 'dev' server in the cohort, ie the one executing the cts
 * graph cts reports test is complete, yet only 218 tests completed out of a possible 218. 1 of these failed.

The exceptions in the log files contain many of:
```
Wed Sep 02 15:06:50 GMT 2020 SUT_Server Types OMRS-AUDIT-0301 The local server has added a new type called SolutionPortDirection with a unique identifier of 4879c96e-26c7-48af-ba92-8277632be733 and a version number of 1 from ODPi Egeria (1.8)
Wed Sep 02 15:06:50 GMT 2020 SUT_Server Exception OMRS-AUDIT-9019 The type definition event processor for the ODPi Egeria (1.8) service caught an unexpected exception java.lang.NullPointerException with message null
Wed Sep 02 15:06:50 GMT 2020 SUT_Server Exception OMRS-AUDIT-9019 Supplementary information: log record id 6bc337c0-ff27-4831-bfda-3d726f1934c6 java.lang.NullPointerException returned message of null and stacktrace of
java.lang.NullPointerException
    at org.odpi.openmetadata.adapters.repositoryservices.graphrepository.repositoryconnector.GraphOMRSMetadataCollection.verifyTypeDef(GraphOMRSMetadataCollection.java:130)
    at org.odpi.openmetadata.repositoryservices.localrepository.repositoryconnector.LocalOMRSMetadataCollection.verifyTypeDef(LocalOMRSMetadataCollection.java:835)
    at org.odpi.openmetadata.repositoryservices.localrepository.repositorycontentmanager.OMRSRepositoryContentManager.processNewTypeDefEvent(OMRSRepositoryContentManager.java:2443)
    at org.odpi.openmetadata.repositoryservices.archivemanager.OMRSArchiveManager.processTypeDefStore(OMRSArchiveManager.java:300)
    at org.odpi.openmetadata.repositoryservices.archivemanager.OMRSArchiveManager.processOpenMetadataArchive(OMRSArchiveManager.java:211)
    at org.odpi.openmetadata.repositoryservices.archivemanager.OMRSArchiveManager.processOpenMetadataTypes(OMRSArchiveManager.java:139)
    at org.odpi.openmetadata.repositoryservices.archivemanager.OMRSArchiveManager.setLocalRepository(OMRSArchiveManager.java:102)
    at org.odpi.openmetadata.repositoryservices.admin.OMRSOperationalServices.initializeCohortMember(OMRSOperationalServices.java:430)
    at org.odpi.openmetadata.adminservices.OMAGServerOperationalServices.activateWithSuppliedConfig(OMAGServerOperationalServices.java:316)
    at org.odpi.openmetadata.adminservices.OMAGServerOperationalServices.activateWithStoredConfig(OMAGServerOperationalServices.java:152)
    at org.odpi.openmetadata.adminservices.spring.OperationalServicesResource.activateWithStoredConfig(OperationalServicesResource.java:59)
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke(Method.java:498)
    at org.springframework.web.method.support.InvocableHandlerMethod.doInvoke(InvocableHandlerMethod.java:190)
    at org.springframework.web.method.support.InvocableHandlerMethod.invokeForRequest(InvocableHandlerMethod.java:138)
    at org.springframework.web.servlet.mvc.method.annotation.ServletInvocableHandlerMethod.invokeAndHandle(ServletInvocableHandlerMethod.java:105)
    at org.springframework.web.servlet.mvc.method.annotation.RequestMappingHandlerAdapter.invokeHandlerMethod(RequestMappingHandlerAdapter.java:878)
    at org.springframework.web.servlet.mvc.method.annotation.RequestMappingHandlerAdapter.handleInternal(RequestMappingHandlerAdapter.java:792)
    at org.springframework.web.servlet.mvc.method.AbstractHandlerMethodAdapter.handle(AbstractHandlerMethodAdapter.java:87)
    at org.springframework.web.servlet.DispatcherServlet.doDispatch(DispatcherServlet.java:1040)
    at org.springframework.web.servlet.DispatcherServlet.doService(DispatcherServlet.java:943)
    at org.springframework.web.servlet.FrameworkServlet.processRequest(FrameworkServlet.java:1006)
    at org.springframework.web.servlet.FrameworkServlet.doPost(FrameworkServlet.java:909)
    at javax.servlet.http.HttpServlet.service(HttpServlet.java:652)
    at org.springframework.web.servlet.FrameworkServlet.service(FrameworkServlet.java:883)
    at javax.servlet.http.HttpServlet.service(HttpServlet.java:733)
    at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:231)
    at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
    at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:53)
    at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
    at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
    at org.springframework.web.filter.RequestContextFilter.doFilterInternal(RequestContextFilter.java:100)
    at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:119)
    at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
    at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
    at org.springframework.web.filter.FormContentFilter.doFilterInternal(FormContentFilter.java:93)
    at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:119)
    at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
    at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
    at org.springframework.boot.actuate.metrics.web.servlet.WebMvcMetricsFilter.doFilterInternal(WebMvcMetricsFilter.java:93)
    at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:119)
    at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
    at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
    at org.springframework.web.filter.CharacterEncodingFilter.doFilterInternal(CharacterEncodingFilter.java:201)
    at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:119)
    at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
    at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
    at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:202)
    at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:96)
    at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:541)
    at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:139)
    at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:92)
    at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:74)
    at org.apache.catalina.valves.RemoteIpValve.invoke(RemoteIpValve.java:747)
    at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:343)
    at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:373)
    at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:65)
    at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:868)
    at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1589)
    at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:49)
    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
    at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61)
    at java.lang.Thread.run(Thread.java:748)
```

cc: @mandy-chessell @grahamwallis

**Comment by planetf1** [comment]
The exception occurs in `graphStore.createEntityIndexes(typeDef);` so perhaps the metadata collection failed to initialise completely.

**Comment by planetf1** [comment]
Repeating the test locally, and the graph repo continues past this point without exception.

My hunch is that the graph repo connector is not loading correctly, probably due to some kind of classpath issue (given the problem found with dependencies & the packaging change) and that failure is not being reported properly....

**Comment by planetf1** [comment]
The cause is not the same as #3451 though there too the server initialization continued despite a critical error

**Comment by planetf1** [comment]
The graph was not created. Enabling logging (should NOT be needed to debug) showed:
```
➜  charts git:(egeria-release-2.2) ✗ cat /tmp/r | grep ERROR
16:12:04.168 [REST:OMAG Server Operational Services:activateWithSuppliedConfig] ERROR o.o.o.a.r.g.r.GraphOMRSGraphFactory - open could not open graph
16:12:04.172 [REST:OMAG Server Operational Services:activateWithSuppliedConfig] ERROR o.o.o.a.r.g.r.GraphOMRSMetadataStore - GraphOMRSMetadataStore Could not open graph database
16:12:04.172 [REST:OMAG Server Operational Services:activateWithSuppliedConfig] ERROR o.o.o.a.r.g.r.GraphOMRSMetadataCollection - GraphOMRSMetadataCollection could not create graph metadata collection for repository name SUT_Server
```
And in context:
```
Wed Sep 02 16:12:02 GMT 2020 SUT_Server Startup OMRS-AUDIT-0001 The Open Metadata Repository Services (OMRS) is initializing the subsystems to support a metadata server
Wed Sep 02 16:12:02 GMT 2020 SUT_Server Startup OMRS-AUDIT-0003 The local repository is initializing the metadata collection named SUT_MDR with an id of 82a7af86-cdd0-4557-a78a-937b6be2f9
05
Wed Sep 02 16:12:02 GMT 2020 SUT_Server Startup OMRS-AUDIT-0029 The local repository outbound event manager is initializing
16:12:04.168 [REST:OMAG Server Operational Services:activateWithSuppliedConfig] ERROR o.o.o.a.r.g.r.GraphOMRSGraphFactory - open could not open graph
16:12:04.172 [REST:OMAG Server Operational Services:activateWithSuppliedConfig] ERROR o.o.o.a.r.g.r.GraphOMRSMetadataStore - GraphOMRSMetadataStore Could not open graph database
16:12:04.172 [REST:OMAG Server Operational Services:activateWithSuppliedConfig] ERROR o.o.o.a.r.g.r.GraphOMRSMetadataCollection - GraphOMRSMetadataCollection could not create graph metadata collection for repository name SUT_Server
Wed Sep 02 16:12:04 GMT 2020 SUT_Server Information OMRS-AUDIT-0050 The Open Metadata Repository Services (OMRS) is about to process open metadata archive Open Metadata Types
```

Thus the same error handling omission found in #3451 applies, and needs addressing - we should see an audit log message, and the server should fail to start.

However it's not clear yet WHY the metadata collection could not be created at this log level...

**Comment by planetf1** [comment]
```
cationFilterChain.java:166)
        at org.springframework.web.filter.CharacterEncodingFilter.doFilterInternal(CharacterEncodingFilter.java:201)
        at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:119)
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
        at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:202)
        at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:96)
        at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:541)
        at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:139)
        at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:92)
        at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:74)
        at org.apache.catalina.valves.RemoteIpValve.invoke(RemoteIpValve.java:747)
        at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:343)
        at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:373)
        at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:65)
        at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:868)
        at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1589)
        at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:49)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61)
        at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.reflect.InvocationTargetException: null
        at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
        at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
        at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
        at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
        at org.janusgraph.util.system.ConfigurationUtil.instantiate(ConfigurationUtil.java:58)
        ... 73 common frames omitted
Caused by: java.lang.UnsatisfiedLinkError: sun.nio.fs.LinuxNativeDispatcher.getmntent(JLsun/nio/fs/UnixMountEntry;)I
        at sun.nio.fs.LinuxNativeDispatcher.getmntent(Native Method)
        at sun.nio.fs.LinuxFileSystem.getMountEntries(LinuxFileSystem.java:85)
        at sun.nio.fs.LinuxFileStore.findMountEntry(LinuxFileStore.java:70)
        at sun.nio.fs.UnixFileStore.<init>(UnixFileStore.java:65)
        at sun.nio.fs.LinuxFileStore.<init>(LinuxFileStore.java:44)
        at sun.nio.fs.LinuxFileSystemProvider.getFileStore(LinuxFileSystemProvider.java:51)
        at sun.nio.fs.LinuxFileSystemProvider.getFileStore(LinuxFileSystemProvider.java:39)
        at sun.nio.fs.UnixFileSystemProvider.getFileStore(UnixFileSystemProvider.java:368)
        at java.nio.file.Files.getFileStore(Files.java:1461)
        at com.sleepycat.je.utilint.Java7FileStoreInfo.<init>(Java7FileStoreInfo.java:56)
        at com.sleepycat.je.utilint.Java7FileStoreInfo$Java7Factory.factoryGetInfo(Java7FileStoreInfo.java:39)
        at com.sleepycat.je.utilint.FileStoreInfo.getInfo(FileStoreInfo.java:144)
        at com.sleepycat.je.cleaner.Cleaner.<init>(Cleaner.java:355)
        at com.sleepycat.je.dbi.EnvironmentImpl.<init>(EnvironmentImpl.java:690)
        at com.sleepycat.je.dbi.EnvironmentImpl.<init>(EnvironmentImpl.java:518)
        at com.sleepycat.je.dbi.DbEnvPool.getEnvironment(DbEnvPool.java:192)
        at com.sleepycat.je.Environment.makeEnvironmentImpl(Environment.java:278)
        at com.sleepycat.je.Environment.<init>(Environment.java:258)
        at com.sleepycat.je.Environment.<init>(Environment.java:228)
        at org.janusgraph.diskstorage.berkeleyje.BerkeleyJEStoreManager.initialize(BerkeleyJEStoreManager.java:117)
        at org.janusgraph.diskstorage.berkeleyje.BerkeleyJEStoreManager.<init>(BerkeleyJEStoreManager.java:87)
        ... 78 common frames omitted
17:05:11.559 [REST:OMAG Server Operational Services:activateWithSuppliedConfig] ERROR o.o.o.a.r.g.r.GraphOMRSMetadataStore - GraphOMRSMetadataStore Could not open graph database
17:05:11.563 [REST:OMAG Server Operational Services:activateWithSuppliedConfig] ERROR o.o.o.a.r.g.r.GraphOMRSMetadataCollection - GraphOMRSMetadataCollection could not create graph metadata collection for repository name SUT_Server
17:05:11.564 [REST:OMAG Server Operational Services:activateWithSuppliedConfig] DEBUG o.o.o.r.a.OMRSArchiveBuilder - Adding PrimitiveDef: PrimitiveDef{name='object', primitiveDefCategory=PrimitiveDefCategory{code=0, name='object', javaClassName='java.lang.Object', guid='1c4b21f4-0b67-41a7-a6ed-2af185eb9b3b'}, category=AttributeTypeDefCategory{typeCode=1, typeName='Primitive', typeDescription='A primitive type.'}, guid='1c4b21f4-0b67-41a7-a6ed-2af185eb9b3b', description='null', descriptionGUID='null'}
```

**Comment by planetf1** [comment]
The cause looks like
```
java.lang.IllegalArgumentException: Could not instantiate implementation: org.janusgraph.diskstorage.berkeleyje.BerkeleyJEStoreManager
```

**Comment by planetf1** [comment]
More testing
 * Graph repo working fine locally run from built chassis directly
 * Graph repo fine from distribution
 * Graph repo fine from distribution with LOADER_PATH set
 * Still fine as above but having ensure the graph does not exist, all config deleted etc
 * fails with docker compose (local)

Trying to identify why we are getting a failure in the container only.

**Comment by planetf1** [comment]
<img width="2327" alt="Screenshot 2020-09-04 at 14 31 10" src="https://user-images.githubusercontent.com/7292002/92245490-5c6b0000-eebc-11ea-8a7e-94da0b2314fc.png">

**Comment by planetf1** [comment]
Setup the k8s environment with debug so I could attach to jvm. The failure is at:
`return new StandardJanusGraph((new GraphDatabaseConfigurationBuilder()).build(configuration));` from 
```
open:161, JanusGraphFactory (org.janusgraph.core)
open:132, JanusGraphFactory (org.janusgraph.core)
open:122, JanusGraphFactory (org.janusgraph.core)
open:261, JanusGraphFactory$Builder (org.janusgraph.core)
open:92, GraphOMRSGraphFactory (org.odpi.openmetadata.adapters.repositoryservices.graphrepository.repositoryconnector)
<init>:134, GraphOMRSMetadataStore (org.odpi.openmetadata.adapters.repositoryservices.graphrepository.repositoryconnector)
<init>:101, GraphOMRSMetadataCollection (org.odpi.openmetadata.adapters.repositoryservices.graphrepository.repositoryconnector)
setMetadataCollectionId:41, GraphOMRSRepositoryConnector (org.odpi.openmetadata.adapters.repositoryservices.graphrepository.repositoryconnector)
setMetadataCollectionId:352, LocalOMRSRepositoryConnector (org.odpi.openmetadata.repositoryservices.localrepository.repositoryconnector)
getLocalOMRSConnector:1131, OMRSOperationalServices (org.odpi.openmetadata.repositoryservices.admin)
initializeLocalRepository:716, OMRSOperationalServices (org.odpi.openmetadata.repositoryservices.admin)
initializeCohortMember:401, OMRSOperationalServices (org.odpi.openmetadata.repositoryservices.admin)
activateWithSuppliedConfig:316, OMAGServerOperationalServices (org.odpi.openmetadata.adminservices)
activateWithStoredConfig:152, OMAGServerOperationalServices (org.odpi.openmetadata.adminservices)
activateWithStoredConfig:59, OperationalServicesResource (org.odpi.openmetadata.adminservices.spring)
invoke0:-1, NativeMethodAccessorImpl (sun.reflect)
invoke:62, NativeMethodAccessorImpl (sun.reflect)
invoke:43, DelegatingMethodAccessorImpl (sun.reflect)
invoke:498, Method (java.lang.reflect)
doInvoke:190, InvocableHandlerMethod (org.springframework.web.method.support)
invokeForRequest:138, InvocableHandlerMethod (org.springframework.web.method.support)
invokeAndHandle:105, ServletInvocableHandlerMethod (org.springframework.web.servlet.mvc.method.annotation)
invokeHandlerMethod:878, RequestMappingHandlerAdapter (org.springframework.web.servlet.mvc.method.annotation)
handleInternal:792, RequestMappingHandlerAdapter (org.springframework.web.servlet.mvc.method.annotation)
handle:87, AbstractHandlerMethodAdapter (org.springframework.web.servlet.mvc.method)
doDispatch:1040, DispatcherServlet (org.springframework.web.servlet)
doService:943, DispatcherServlet (org.springframework.web.servlet)
processRequest:1006, FrameworkServlet (org.springframework.web.servlet)
doPost:909, FrameworkServlet (org.springframework.web.servlet)
service:652, HttpServlet (javax.servlet.http)
service:883, FrameworkServlet (org.springframework.web.servlet)
service:733, HttpServlet (javax.servlet.http)
internalDoFilter:231, ApplicationFilterChain (org.apache.catalina.core)
doFilter:166, ApplicationFilterChain (org.apache.catalina.core)
doFilter:53, WsFilter (org.apache.tomcat.websocket.server)
internalDoFilter:193, ApplicationFilterChain (org.apache.catalina.core)
doFilter:166, ApplicationFilterChain (org.apache.catalina.core)
doFilterInternal:100, RequestContextFilter (org.springframework.web.filter)
doFilter:119, OncePerRequestFilter (org.springframework.web.filter)
internalDoFilter:193, ApplicationFilterChain (org.apache.catalina.core)
doFilter:166, ApplicationFilterChain (org.apache.catalina.core)
doFilterInternal:93, FormContentFilter (org.springframework.web.filter)
doFilter:119, OncePerRequestFilter (org.springframework.web.filter)
internalDoFilter:193, ApplicationFilterChain (org.apache.catalina.core)
doFilter:166, ApplicationFilterChain (org.apache.catalina.core)
doFilterInternal:93, WebMvcMetricsFilter (org.springframework.boot.actuate.metrics.web.servlet)
doFilter:119, OncePerRequestFilter (org.springframework.web.filter)
internalDoFilter:193, ApplicationFilterChain (org.apache.catalina.core)
doFilter:166, ApplicationFilterChain (org.apache.catalina.core)
doFilterInternal:201, CharacterEncodingFilter (org.springframework.web.filter)
doFilter:119, OncePerRequestFilter (org.springframework.web.filter)
internalDoFilter:193, ApplicationFilterChain (org.apache.catalina.core)
doFilter:166, ApplicationFilterChain (org.apache.catalina.core)
invoke:202, StandardWrapperValve (org.apache.catalina.core)
invoke:96, StandardContextValve (org.apache.catalina.core)
invoke:541, AuthenticatorBase (org.apache.catalina.authenticator)
invoke:139, StandardHostValve (org.apache.catalina.core)
invoke:92, ErrorReportValve (org.apache.catalina.valves)
invoke:74, StandardEngineValve (org.apache.catalina.core)
invoke:747, RemoteIpValve (org.apache.catalina.valves)
service:343, CoyoteAdapter (org.apache.catalina.connector)
service:373, Http11Processor (org.apache.coyote.http11)
process:65, AbstractProcessorLight (org.apache.coyote)
process:868, AbstractProtocol$ConnectionHandler (org.apache.coyote)
doRun:1589, NioEndpoint$SocketProcessor (org.apache.tomcat.util.net)
run:49, SocketProcessorBase (org.apache.tomcat.util.net)
runWorker:1149, ThreadPoolExecutor (java.util.concurrent)
run:624, ThreadPoolExecutor$Worker (java.util.concurrent)
run:61, TaskThread$WrappingRunnable (org.apache.tomcat.util.threads)
run:748, Thread (java.lang)
```

**Comment by planetf1** [comment]
Cannot debug any further -- this shows why:
<img width="1226" alt="Screenshot 2020-09-04 at 14 44 57" src="https://user-images.githubusercontent.com/7292002/92246184-4f9adc00-eebd-11ea-9815-9313ab431132.png">

**Comment by planetf1** [comment]
This actually looks like it could be a jdk issue which may explain why we work ok locally -- even using the assembly, but are specifically failing in the docker image.

Our docker image is at https://github.com/odpi/egeria/blob/master/open-metadata-resources/open-metadata-deployment/docker/egeria/Dockerfile and is based on `openjdk:8-jre-alpine` which is a popular base for docker images

**Comment by planetf1** [comment]
JDk being used in alpine is
```
bash-4.4$ java -version
openjdk version "1.8.0_212"
OpenJDK Runtime Environment (IcedTea 3.12.0) (Alpine 8.212.04-r0)
OpenJDK 64-Bit Server VM (build 25.252-b09, mixed mode)
```

**Comment by planetf1** [comment]
It looks as if these images are maintained by the docker community but once released do not get updated -- mostly only the latest java version (for example 16-jdk-alpine) is being updated -- so we're somewhat back level / missing security fixes etc.

Probably should look at an alternate image. The regular jre is fresher (last month) for both java 8 and 11, though unsure on basis.  The closest is probably `11-jre-slim' https://hub.docker.com/layers/openjdk/library/openjdk/11-jre-slim/images/sha256-ae3117ed700caa58541fbf93a4ca9f5a76088ad69fef250424dad4d85f1b40f8?context=explore

Will experiment to see if the hunch on a jdk issue is correct. The original issue felt like a classpath issue - but we're into native methods so it's a native library link issue.. but not where I expected .. seems to be in the jdk native libraries rather than anything to do with janusgraph or egeria.

Unclear why this recently stopped working however, unless it was caused by a change in janus or berkeley.

**Comment by planetf1** [comment]
Supported Java ubi images... more at https://developers.redhat.com/blog/2020/06/25/introducing-the-red-hat-build-of-the-openjdk-universal-base-images-now-in-red-hat-enterprise-linux-8-2/

**Comment by planetf1** [comment]
I've now raised 2 PRs for 2.2 & master.

It seems as if my hunch was correct. Unfortunately the openjdk team did not create jre alpine images for java 11, and the version 8 code seems too old to correctly work with janusgraph -- the jre is quite a few builds back.

Java 14 is the latest supported release and seems to work. Not quite ideal as technically we support java 8, 11 - the LTS releases. Unfortunately the openjdk images didn't really deliver the updates we expected (we do try and update from the alpine repo when the image is built... but there are no later updates). It seems as if only current versions are built.

Moving forward we should switch to UBI images as mentioned here. I will open a new issue to track this. I decided not
to make the switch for the immediate fix, as it is a different distro (redhat) which means different packages that may be prereqs, different commands to update (yum) and a little more testing needed -- as well as ensuring we switch and then really are compliant.
